gnssanalysis/gn_io/sp3.py

```diff
@@ -477,28 +477,28 @@ def getVelPoly(sp3Df: _pd.DataFrame, deg: int = 35) -> _pd.DataFrame:
 
     """
     est = sp3Df.unstack(1).EST[["X", "Y", "Z"]]
-    x = est.index.get_level_values("J2000").values
-    # x = est.index.values
-    y = est.values
+    times = est.index.get_level_values("J2000").values
+    positions = est.values
 
-    off, scl = mapparm([x.min(), x.max()], [-1, 1])  # map from input scale to [-1,1]
+    # map from input scale to [-1,1]
+    offset, scale_factor = mapparm([times.min(), times.max()], [-1, 1])
 
-    x_new = off + scl * (x)
-    coeff = _np.polyfit(x=x_new, y=y, deg=deg)
+    normalised_times = offset + scale_factor * (times)
+    coeff = _np.polyfit(x=normalised_times, y=positions, deg=deg)
 
-    x_prev = off + scl * (x - 1)
-    x_next = off + scl * (x + 1)
+    time_prev = offset + scale_factor * (times - 1)
+    time_next = offset + scale_factor * (times + 1)
```

Comment:
Should this actually be called `times_next`? Given it appears to be *all* the time values, shifted by one step forward.
@seballgeyer

Reply:
More a look into this function, obscurer it looks... 
it seems that the 20 lines functions is doing something that numpy can do by itself in 2 or 3 function calls. 

I would suggest to merge this PR and eventually rewrite this function later.